File: leetcode/c++/5 Longest Palindromic Substring.cpp
class Solution {
public:
    void search(string s, int left, int right, int& index, int& maxLen){
        while(left >= 0 && right < s.length() && s[left] == s[right]){
            ++right;
            --left;
        }
        if(maxLen < right - left - 1){
            index = left + 1;
            maxLen = right - left - 1;
        }
    }
    
    string longestPalindrome(string s) {
        if(s.size() <= 1){
            return s;
        }
        int index = 0;
        int maxLen = 0;
        for (int i = 0; i < s.size() - 1; ++i) {
            search(s, i, i, index, maxLen);
            search(s, i, i + 1, index, maxLen);
        }
        return s.substr(index, maxLen);
    }
};

File: leetcode/ts/5. Longest Palindromic Substring.ts
function longestPalindrome(s: string): string {
    if(s.length < 2){
        return s
    }
    let str : string = ''
    let left : number = 0
    let right : number = 0
    let start : number = 0
    let maxLen : number = 0
    
    for(let i = 0; i < s.length; ++i){
        for(let j = 0; j < 2; ++j){
            left = i;
            right = i + j;
            while(left >= 0 && right < s.length && s[left--] == s[right++]){
                if(right - left - 1 > maxLen){
                    start = left + 1
                    maxLen = right - left - 1
                }
            }
        }
    }
    return s.slice(start, start + maxLen)
};